BnmaAnalysis.R

```diff
@@ -0,0 +1,186 @@
+#' @param dataFormat "long" or "wide"
+#' @param outcomeType "continuous" or "binomial"
+#' @param ref An element of 'BRdata$T', the reference treatment
+#' @return A list with elements 'ArmLevel' and 'Treat.order'
+#'          'ArmLevel' is a data frame containing 'Study', 'Treat', 'N', 'Outcomes',
+#'          and (for outcomeType="continuous") 'SD'
+FormatForBnma <- function(BRdata, dataFormat, outcomeType, ref){
+  if(dataFormat == "wide"){
+    BRdata2 <- as.data.frame(WideToLong(BRdata, ConBi=outcomeType))
+  } else if(dataFormat == "long"){
+    BRdata2 <- BRdata
+  }
+  #Treatment order (put 'ref' first)
+  Treat.order <- c(ref, unique(BRdata2$T)[-which(unique(BRdata2$T) == ref)])
+  #Arm-level data
+  if(outcomeType == "binomial"){
+    ArmLevel <- dplyr::rename(BRdata2, "Treat"="T", "Outcomes"="R")
+  } else if(outcomeType == "continuous"){
+    ArmLevel <- dplyr::rename(BRdata2, "Treat"="T", "Outcomes"="Mean")
+  }
+  return(list(ArmLevel=ArmLevel, Treat.order=Treat.order))
+}
+
+#Examples
+BRbinBnma <- FormatForBnma(BRdata=BRdataBin, dataFormat="long", outcomeType="binomial", ref="Placebo")
+BRbinBnma_wide <- FormatForBnma(BRdata=BRdataBin_wide, dataFormat="wide", outcomeType="binomial", ref="Placebo")
+BRcontBnma <- FormatForBnma(BRdata=BRdataCont, dataFormat="long", outcomeType="continuous", ref="Placebo")
+
+
+
+
+#' Fits the baseline risk meta-regression model in BNMA
+#'
+#' @param BRdata A list of data in the format produced by the 'FormatForBnma' function
+#' @param outcomeType "continuous" or "binomial"
+#' @param effectsType "fixed" or "random"
+#' @param covParameters "common", "exchangable", or "independent"
+#' @return Output from bnma::network.run
+BaselineRiskRegression <- function(BRdata, outcomeType, effectsType, covParameters){
+  if(outcomeType == "binomial"){
+    BRnetwork <- with(BRdata, bnma::network.data(Outcomes = ArmLevel$Outcomes,
+                                                 Study = ArmLevel$Study,
+                                                 Treat = ArmLevel$Treat,
+                                                 N = ArmLevel$N,                                          
+                                                 response = "binomial",
+                                                 Treat.order = Treat.order,
+                                                 type = effectsType,
+                                                 baseline = covParameters,
+                                                 baseline.risk = "independent"))
+  }
+  if(outcomeType == "continuous"){
+    BRnetwork <- with(BRdata, bnma::network.data(Outcomes = ArmLevel$Outcomes,
+                                                 Study = ArmLevel$Study,                                           
+                                                 Treat = ArmLevel$Treat,
+                                                 response = "normal",
+                                                 SE = ArmLevel$SD,

```

Comment:
Aren't standard error and standard deviation different values? Should there be some transformation?

Reply:
@BalancedmonkeY Good point. I think I've had a misunderstanding for a long time about this. In meta-analysis language people generally use variance to mean both SE^2 and SD^2. I have also been using SE and SD interchangeably because I thought they were all SEs, because I thought they were the output from a model rather than raw data. Am I right in saying the y's and their SDs are actually raw data rather than the output from a model? In which case the SDs in MetaInsight data really are SDs and not SEs. In my simulated data I called them SDs but really they are SEs. So the upshot is yes, the function can take SE or the pair (SD, N), and it should be SD and N.

Reply:
Yeah, I often get confused between them all as well!
But yeah, it's my understanding that the mean and SD are raw data from each study -> but how each study obtained that information I'm not 100% sure of. 
The fact that on the data upload page of MetaInsight it says "SD contains the standard deviation of the outcome in each arm of the study", I think we can assume there are SD, rather than SE.
So in conclusion, you're change is good :)